include/alloc-inl.h

```diff
@@ -30,12 +30,13 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <stddef.h>
 
 #include "config.h"
 #include "types.h"
 #include "debug.h"
 
-/* Initial size used for ck_maybe_grow */
+/* Initial size used for maybe_grow */
 #define INITIAL_GROWTH_SIZE (64)
 
```

Comment:
the ->size value in the structure could either be the full size of the allocated memory, or the size available to the user.
currently it is the previous.

if ->size is just internal it does not matter and whatever results in the fewest computations required should be it.

if this is given to the user it should be the latter as otherwise a user has to know do deduct a pointer size from the value and forgetting that would lead to overruns.